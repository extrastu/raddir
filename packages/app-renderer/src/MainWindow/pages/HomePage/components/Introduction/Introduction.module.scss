.introduction {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 60vh;
    padding-top: 20vh;
    font-size: calc(10px + 2vmin);
}

.link {
    color: #61dafb;
}

.logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .logo {
        animation: logo-spin infinite 20s linear;
    }
}

@keyframes logo-spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

.appDetails {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-end;
    padding: 0 8px 8px;
    font-size: calc(5px + 1vmin);
}
