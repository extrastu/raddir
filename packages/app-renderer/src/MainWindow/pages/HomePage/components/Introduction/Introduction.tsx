import type { AppDetails } from '@tecra/app-common';

import { Assets } from 'src/assets';

import css from './Introduction.module.scss';

interface IntroductionProps {
    appDetails: AppDetails | null;
}

export function Introduction({ appDetails }: IntroductionProps): JSX.Element {
    return (
        <>
            <header className={css.introduction}>
                <img className={css.logo} src={Assets.logo} alt="logo" />
                <p>
                    Edit <code>src/MainWindow/components/Introduction/Introduction.tsx</code> and save to reload.
                </p>
                <a
                    className={css.link}
                    href="https://github.com/YSoftwareRepo/template-electron-cra"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn template-electron-cra
                </a>
            </header>
            {appDetails && (
                <div className={css.appDetails}>
                    <code>Name: {appDetails.name}</code>
                    <code>Version: {appDetails.version}</code>
                    <code>Electron: {appDetails.module.electron}</code>
                    <code>Chrome: {appDetails.module.chrome}</code>
                    <code>Node.js: {appDetails.module.node}</code>
                    <code>V8: {appDetails.module.v8}</code>
                </div>
            )}
        </>
    );
}
