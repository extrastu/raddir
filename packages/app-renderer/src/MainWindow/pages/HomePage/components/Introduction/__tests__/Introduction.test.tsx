import type { AppDetails } from '@tecra/app-common';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Introduction } from '../Introduction';

describe(`Test component \`${Introduction.name}\``, () => {
    it('should render', () => {
        const appDetails: AppDetails = {
            name: 'tecra',
            version: 'unknown',
            module: {
                chrome: 'unknown',
                electron: 'unknown',
                node: 'unknown',
                v8: 'unknown',
            },
        };

        const element = <Introduction appDetails={appDetails} />;

        expect(renderer.create(element).toJSON()).toMatchSnapshot();

        const { getByText } = render(element);
        const linkElement = getByText(/Learn template-electron-cra/i);
        expect(linkElement).toBeInTheDocument();
    });
});
