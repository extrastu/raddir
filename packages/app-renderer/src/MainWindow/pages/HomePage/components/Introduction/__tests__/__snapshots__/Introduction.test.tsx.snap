// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component \`Introduction\` should render 1`] = `
[
  <header
    className="introduction"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/MainWindow/components/Introduction/Introduction.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="link"
      href="https://github.com/YSoftwareRepo/template-electron-cra"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn template-electron-cra
    </a>
  </header>,
  <div
    className="appDetails"
  >
    <code>
      Name: 
      tecra
    </code>
    <code>
      Version: 
      unknown
    </code>
    <code>
      Electron: 
      unknown
    </code>
    <code>
      Chrome: 
      unknown
    </code>
    <code>
      Node.js: 
      unknown
    </code>
    <code>
      V8: 
      unknown
    </code>
  </div>,
]
`;
